cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -236,6 +235,8 @@ public Void call() throws Exception {
     } catch (Exception e) {
       LOG.warn("Error while deleting namespace {}", namespaceId, e);
       throw new NamespaceCannotBeDeletedException(namespaceId, e);
+    } finally {
+      authorizerInstantiator.get().revoke(namespace);
```

Comment:
Its there right at the top of this method

``// TODO: CDAP-870, CDAP-1427: Delete should be in a single transaction.``